cmd/ci-operator/main.go

```diff
@@ -398,13 +406,30 @@ func (o *options) Run() error {
 		secretName = o.sshSecret.Name
 	}
 	// load the graph from the configuration
-	buildSteps, postSteps, err := defaults.FromConfig(o.configSpec, o.jobSpec, o.templates, o.writeParams, o.artifactDir, o.promote, o.clusterConfig, o.targets.values, dryLogger, secretName)
+	buildSteps, postSteps, err := defaults.FromConfig(o.configSpec, o.jobSpec, o.templates, o.writeParams, o.artifactDir, o.promote, o.clusterConfig, o.leaseClient, o.targets.values, dryLogger, secretName)
 	if err != nil {
 		return fmt.Errorf("failed to generate steps from config: %v", err)
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
-
+	if o.leaseClient != nil {
+		defer func() {
+			if err := o.leaseClient.ReleaseAll(); err != nil {
+				log.Printf("failed to release remaining leases: %v", err)
+			}
+		}()
+		go func() {
+			t := time.NewTicker(30 * time.Second)
+			for {
+				select {
+				case <-ctx.Done():
+					t.Stop()
+				case <-t.C:
+					o.leaseClient.Heartbeat()
```

Comment:
I thought there was a blocking acquire call that we could use?

Reply:
This is only responsible for updating existing leases. The acquire process is done by `lease.Client.Acquire`.